typings/index.d.ts

```diff
@@ -731,6 +732,15 @@ declare module 'discord.js' {
     public toJSON(): object;
   }
 
+  export class GuildBan {
+    constructor(client: Client, data: object, guild: Guild);
+    public readonly guild: Guild;
+    public readonly user: User;
+    public readonly partial: boolean;
+    public readonly reason?: string;
```

Comment:
```suggestion
    public readonly reason: string | null;
```
based on
```ts
    /**
     * The reason for the ban
     * @type {?string}
     */
    this.reason = null;
```

Reply:
it should actually be
```suggestion
    public readonly reason?: string | null;
```
as currently its set to `undefined` if its partial

Reply:
solved in bb141f4